### PR TITLE
feat: new swap events and properties in taxonomy 

### PR DESCRIPTION
**Pull request by lynnshaoyu**
1. New Event: Wrap Token Transaction Submitted
Screen Shot 2022-07-28 at 1 16 02 AM

2.  New Event: Approve Token Transaction Submitted
Screen Shot 2022-07-28 at 1 29 57 AM
@willhennessy the approve token can come from multiple places in the app though, do we want to log all? currently im just logging the ones coming from swap page. 

jira: https://uniswaplabs.atlassian.net/browse/WEB-623?atlOrigin=eyJpIjoiNDY1YTM0NzVjMWI1NDQ3ZjhjY2U2NjYzZDY2NThkOTYiLCJwIjoiaiJ9

### CODE REVIEW

**Comment by aadams** [comment]
The routing data look like exactly what we wanted! Is there anyway that I could see a multi-hop trade just to make sure?

File: src/pages/Swap/index.tsx [hunk 1]
```diff
@@ -643,53 +688,67 @@ export default function Swap() {
...
-                        altDisabledStyle={approvalState === ApprovalState.PENDING} // show solid button while waiting
-                        confirmed={
-                          approvalState === ApprovalState.APPROVED || signatureState === UseERC20PermitState.SIGNED
+                      <TraceEvent
+                        events={[Event.onClick]}
+                        name={EventName.APPROVE_TOKEN_TXN_SUBMITTED}
+                        element={ElementName.APPROVE_TOKEN_BUTTON}
+                        properties={formatApproveTokenTxnSubmittedEventProperties(approvalOptimizedTrade)}
+                        shouldLogImpression={
+                          !(
```
Comment (vm) [inline src/pages/Swap/index.tsx:697]: i find this conditional kinda confusing. maybe you can name it as a variable? basically what is the stuff inside the () and why should you only log if that is false

File: src/pages/Swap/index.tsx [hunk 2]
```diff
@@ -625,15 +658,27 @@ export default function Swap() {
...
+                  <TraceEvent
+                    events={[Event.onClick]}
+                    name={EventName.WRAP_TOKEN_TXN_SUBMITTED}
+                    element={ElementName.WRAP_TOKEN_BUTTON}
+                    properties={formatWrapTokenTxnSubmittedEventProperties(
+                      currencies[Field.INPUT],
+                      currencies[Field.OUTPUT],
+                      parsedAmount
+                    )}
+                    shouldLogImpression={!Boolean(wrapInputError)}
```
Comment (vm) [inline src/pages/Swap/index.tsx:670]: can you check `!error`?

File: src/pages/Swap/index.tsx [hunk 3]
```diff
@@ -92,7 +92,40 @@ export function getIsValidSwapQuote(
...
+    chain_id: approvalOptimizedTrade.inputAmount.currency.chainId,
+    token_symbol: approvalOptimizedTrade.inputAmount.currency.symbol,
+    token_address: approvalOptimizedTrade.inputAmount.currency.isNative
+      ? NATIVE_CHAIN_ADDRESS
+      : approvalOptimizedTrade.inputAmount.currency.address,
+  }
+}
+
+const formatWrapTokenTxnSubmittedEventProperties = (
+  inputCurrency: Currency | null | undefined,
```
Comment (vm) [inline src/pages/Swap/index.tsx:113]: why can it be `null` or `undefined`?

File: src/pages/Swap/index.tsx [hunk 4]
```diff
@@ -92,7 +92,40 @@ export function getIsValidSwapQuote(
...
+  outputCurrency: Currency | null | undefined,
+  parsedAmount: CurrencyAmount<Currency> | undefined
+) => {
+  if (!inputCurrency || !outputCurrency || !parsedAmount) return {}
+  return {
+    token_in_address: inputCurrency.isNative ? NATIVE_CHAIN_ADDRESS : inputCurrency.address,
+    token_out_address: outputCurrency.isNative ? NATIVE_CHAIN_ADDRESS : outputCurrency.address,
+    token_in_symbol: inputCurrency.symbol,
+    token_out_symbol: outputCurrency.symbol,
+    chain_id: inputCurrency.chainId === outputCurrency.chainId ? inputCurrency.chainId : undefined,
```
Comment (vm) [inline src/pages/Swap/index.tsx:123]: how could the `chain_ids` not be the same? feels like that should be handled before here

Comment (vm) [reply src/pages/Swap/index.tsx:115]: i think fine to avoid this because you shoudn't need to update the swap code.

File: src/components/swap/SwapModalFooter.tsx [hunk 5]
```diff
@@ -34,6 +37,39 @@ interface AnalyticsEventProps {
   tokenOutAmountUsd: string | undefined
   lpFeePercent: Percent
   swapQuoteReceivedDate: Date | undefined
+  routes: RoutingDiagramEntry[]
+}
+
+const formatRoutesEventProperties = (routes: RoutingDiagramEntry[]) => {
```
Comment (vm) [inline src/components/swap/SwapModalFooter.tsx:43]: i think if you do the `forEach` as a single loop and skip storing the intermediate variables like `routesPercentages` this would be easier to read.

i'm not stuck on that and it's up to you, but i find this a bit hard to follow right now.

File: src/pages/Swap/index.tsx (hunk 1 above)
Comment (lynnshaoyu) [reply src/pages/Swap/index.tsx:685]: yeah good call thanks. its basically saying dont log when the approve button is disabled.

File: src/pages/Swap/index.tsx (hunk 4 above)
Comment (lynnshaoyu) [reply src/pages/Swap/index.tsx:115]: true, removed check.

File: src/pages/Swap/index.tsx (hunk 2 above)
Comment (lynnshaoyu) [reply src/pages/Swap/index.tsx:658]: no, this is just the inverse of whats inside disabled={} on line 664.

File: src/pages/Swap/index.tsx (hunk 3 above)
Comment (lynnshaoyu) [reply src/pages/Swap/index.tsx:105]: currencies[Field.INPUT] and currencies[Field.OUTPUT] can both be null or undefined based on state of this react component. currencies is from the hook useDerivedSwapInfo().

File: src/components/swap/SwapModalFooter.tsx (hunk 5 above)
Comment (lynnshaoyu) [reply src/components/swap/SwapModalFooter.tsx:43]: nice catch youre totally right, only one foreach is needed. done.

**Review by vm — APPROVED** [review]

**Review by aadams — APPROVED** [review]
the route data looks good! I think the only thing we would want more is the pool the swap went through, but we can back out that information from the data saved.

**Comment by lynnshaoyu** [comment]
> the route data looks good! I think the only thing we would want more is the pool the swap went through, but we can back out that information from the data saved.

as per discussion offline, i'm removing the routing data for now in this PR. separated out that code into a separate pr https://github.com/Uniswap/interface/pull/4221 we can commit later on pending decisions.
